### PR TITLE
DynamoDB support (WIP)

### PR DESCRIPTION
**Pull request by Idokah**

### CODE REVIEW

File: packages/external-db-config/lib/factory.js [hunk 1]
```diff
@@ -58,6 +64,9 @@ const create = () => {
         case 'airtable':
           internalConfigReader = new gcp.GcpAirtableConfigReader()
           break;
+        case 'dynamo':
+          internalConfigReader = new aws.AwsDynamoConfigReader(region) 
+          break
```
Comment (noam-almog) [inline packages/external-db-config/lib/factory.js:69]: gcp doesn't have dynamo

File: packages/external-db-config/lib/factory.js [hunk 2]
```diff
@@ -14,8 +14,14 @@ const create = () => {
...
 
     case 'aws':
-      internalConfigReader = new aws.AwsConfigReader(secretId || DefaultSecretId, region)
-      break;
+      switch(type) {
+        case 'dynamo':
+          internalConfigReader = new aws.AwsDynamoConfigReader(region) 
+          break
+        default:
+          internalConfigReader = new aws.AwsConfigReader(secretId || DefaultSecretId, region)
```
Comment (noam-almog) [inline packages/external-db-config/lib/factory.js:22]: let's rename the AwsConfigReader to something else

File: packages/external-db-dynamo/lib/dynamo_data_provider.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    constructor(client, filterParser) {
+        this.filterParser = filterParser
+        this.client = client
+        this.docClient = DynamoDBDocument.from(client)
+
+    }
+
+    async find(collectionName, filter, sort, skip, limit) {
+        const {filterExpr} = this.filterParser.transform(filter)
+        const response = await this.docClient
```
Comment (noam-almog) [inline packages/external-db-dynamo/lib/dynamo_data_provider.js:15]: it's better to write
```
const { Items } = ....
```

File: packages/external-db-dynamo/lib/dynamo_data_provider.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                                          ...filterExpr,
+                                          Limit:limit
+                                          })
+        return response.Items.map(patchFixDates)
+    }
+
+    async count(collectionName, filter) {
+        const {filterExpr} = this.filterParser.transform(filter)
+        const response = await this.docClient
+                         .scan({TableName: collectionName,
```
Comment (noam-almog) [inline packages/external-db-dynamo/lib/dynamo_data_provider.js:26]: let's discuss the difference between scan and query

File: packages/external-db-config/lib/factory.js (hunk 1 above)
Comment (Idokah) [reply packages/external-db-config/lib/factory.js:69]: it's not in gcp, it's in azure for the e2e test. 
` process.env.CLOUD_VENDOR = 'azr'`   (all e2e tests are on azure)

**Pull request merged**